### PR TITLE
Improve Astro Configs for build

### PR DESCRIPTION
**Pull request by ManuelGil**
## Added

- Add [@astrojs/prefetch](https://docs.astro.build/en/guides/integrations-guide/prefetch/) dependency
- Add Robots configs
- Add Sitemap configs

## Changed

- Upgrade and sort dependencies

## Fixed

- Fix localhost port in site config

### CODE REVIEW

File: public/robots.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+User-agent: *
```
Comment (gndx) [inline public/robots.txt:1]: This file is autogenerated, please delete.

File: public/site.webmanifest [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (gndx) [inline public/site.webmanifest:1]: What's the purpose of this file?

File: public/sitemap.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (gndx) [inline public/sitemap.xml:1]: This file is autogenerated, please delete.

File: src/env.d.ts [hunk 4]
```diff
@@ -1 +1,2 @@
+/// <reference path="../.astro/types.d.ts" />
```
Comment (gndx) [inline src/env.d.ts:1]: ???

File: src/layouts/Base.astro [hunk 5]
```diff
@@ -64,6 +64,10 @@ const ogImage = createOgImage({
       name="description"
       content={description ? description : config.metadata.meta_description}
     />
+    <!-- Sitemap -->
+    <link rel="sitemap" href="/sitemap-index.xml" />
+    <!-- Manifest -->
+    <link rel="manifest" href="/site.webmanifest" />
```
Comment (gndx) [inline src/layouts/Base.astro:70]: This line is a duplicate and is auto-generated.

**Review by gndx — CHANGES_REQUESTED** [review]

File: src/env.d.ts (hunk 4 above)
Comment (ManuelGil) [reply src/env.d.ts:1]: This line was autogenerated when 'prefetch' dependency was added

File: src/layouts/Base.astro (hunk 5 above)
Comment (ManuelGil) [reply src/layouts/Base.astro:70]: Done

File: public/robots.txt (hunk 1 above)
Comment (ManuelGil) [reply public/robots.txt:1]: Done

File: public/sitemap.xml (hunk 3 above)
Comment (ManuelGil) [reply public/sitemap.xml:1]: Done

File: src/layouts/Base.astro (hunk 5 above)
Comment (ManuelGil) [reply src/layouts/Base.astro:70]: The manifest file (site.webmanifest) is called in the line # 70. I'ill revert this change and delete the sitemap line.

Comment (ManuelGil) [reply src/layouts/Base.astro:70]: Sorry the duplicate line has been deleted in another PR.

**Comment by ManuelGil** [comment]
changes have been made

**Review by gndx — APPROVED** [review]

**Pull request closed without merging**